lib/plugins/extension/helper/extension.php

```diff
@@ -915,23 +952,38 @@ public function installArchive($file, $overwrite=false, $base = '') {
                 continue;
             }
 
+            // gather info
             $action = file_exists($target) ? 'update' : 'install';
+            $extname = $item['base'];
+            if($item['type'] == 'template') {
+                $extname = 'template:'.$extname;
+            }
 
-            // copy action
-            if($this->dircopy($item['tmp'], $target)) {
-                // return info
-                $id = $item['base'];
-                if($item['type'] == 'template') {
-                    $id = 'template:'.$id;
+            // prepare plugin hook
+            $extension = new helper_plugin_extension_extension();
+            $extension->setExtension($extname);
+            $data = array(
+                'extension' => $extension,
+                'action' => $action,
+                'src' => $item['tmp'],
+                'dst' => $target
+            );
+
+            // execute the copy
+            $trigger = new Doku_Event('PLUGIN_EXTENSION_CHANGE', $data);
+            if($trigger->advise_before()) {
+                // copy action
+                if($this->dircopy($item['tmp'], $target)) {
+                    $installed_extensions[$extname] = array(
+                        'base' => $item['base'],
+                        'type' => $item['type'],
+                        'action' => $action
+                    );
+                } else {
+                    throw new Exception(sprintf($this->getLang('error_copy') . DOKU_LF, '<bdi>' . $item['base'] . '</bdi>'));
                 }
-                $installed_extensions[$id] = array(
-                    'base' => $item['base'],
-                    'type' => $item['type'],
-                    'action' => $action
-                );
-            } else {
-                throw new Exception(sprintf($this->getLang('error_copy').DOKU_LF, '<bdi>'.$item['base'].'</bdi>'));
             }
+            $trigger->advise_after();
         }
```

Comment:
Here, the result of the event is completely ignored, i.e. the event handler must display the error message, while for `enable` and `disable` the extension manager displays the result. For installation, errors are handled using exceptions in the extension manager, therefore it might make sense to throw an exception with the result? Personally, I don't like how we mix error handling using return values and exceptions in the extension manager, but I guess that is a completely different issue. 

Note that in general here a plugin handling the event could also just throw an exception (in all variants of this event) and the message of the exception would be shown to the user but this is quite different to all/most (?) other events in DokuWiki.